src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -689,7 +689,7 @@ private async Task<bool> PushTargets(IList<TargetSpecification> targets, TargetE
                         }
 
                         // We are already building this target on this request. That's a circular dependency.
-                        ProjectErrorUtilities.ThrowInvalidProject(targetLocation, "CircularDependency", targetSpecification.TargetName);
```

Comment:
Would it make sense to log the parent graph of these, too? Maybe log what it currently logs at normal verbosity and log the parents at diagnostic verbosity?

Reply:
I did some investigation here. I think it is hard to get the parent graph here. 
Here are my reasons:
1. We use `_requestEntry.RequestConfiguration.ActivelyBuildingTargets` to store the targets are being executed. If the current target is in this dictionary, which means we can't find the circular by checking the current `parentTargetEntry`. 
2. Since the target is in `_requestEntry.RequestConfiguration.ActivelyBuildingTargets`, this also means we have lost the previous parent information to add it to `_requestEntry.RequestConfiguration.ActivelyBuildingTargets`.

Reply:
I'm not sure I understand the reason you gave to not use parentTargetEntry. You're saying that we already would have checked for a cycle via parentTargetEntry, so if it gets here, that way to figure out the cycle won't work? Or are you saying the parentTargetEntry is often null?

If there is no way to get the parent information, then I think this is good.

Reply:
I mean "we already would have checked for a cycle via parentTargetEntry, so if it gets here, that way to figure out the cycle won't work". So I think currently we have no way to get the parent information.